### PR TITLE
Fix issue #2

### PR DESCRIPTION
**Pull request by munroesj52**
Avoid the ICE for GCC5/6 by splitting test_4b into two smaller functions.

### CODE REVIEW

**Comment by ThinkOpenly** [comment]
I'm still learning here, but what I see in pull #4 are two commits, one titled "../COMMIT_EDITMSG-20180126", which has already been committed, and a second, "￼For now we should just split arith128_test_i128.c (test_4b) into two".

I recommend creating new branches for your changes, creating pull requests for the branches, deleting the branches, then catching your master up (git fetch ...; git merge) after that pull request has been committed.

File: src/testsuite/arith128_test_i128.c [hunk 1]
```diff
@@ -885,6 +885,20 @@ test_4b (void)
 #endif
   rc += check_vint256 ("vec_mul10ecuq:", l, k, (vui128_t) ec, (vui128_t) e);
 
+  return (rc);
+}
+
+
+/* Needed to split these tests into a separate function to avoid a ICE
+   in GCC 6.3 / AT10.  */
```
Comment (ThinkOpenly) [inline src/testsuite/arith128_test_i128.c]: Do we know why splitting avoids the problem?

File: src/testsuite/arith128_test_i128.c [hunk 2]
```diff
@@ -985,7 +999,6 @@ test_4b (void)
 #endif
   rc += check_vint256 ("vec_mul10ecuq:", l, k, (vui128_t) ec, (vui128_t) e);
 
-#if 1
```
Comment (ThinkOpenly) [inline src/testsuite/arith128_test_i128.c]: This is more than splitting, so why is this change included?

File: src/testsuite/arith128_test_i128.h [hunk 3]
```diff
@@ -54,3 +57,4 @@ extern int
 test_43 (void);
 
 #endif /* TEST_ARITH128_TEST_I128_H_ */
+
```
Comment (ThinkOpenly) [inline src/testsuite/arith128_test_i128.h]: it's preferred to avoid unnecessary whitespace changes

File: src/testsuite/pveclib_test.c [hunk 4]
```diff
@@ -76,3 +77,4 @@ main (void)
     printf ("%d failures reported\n", rc);
   return (rc);
 }
+
```
Comment (ThinkOpenly) [inline src/testsuite/pveclib_test.c]: unnecessary whitespace change here, too

**Comment by munroesj52** [comment]
On Wed, 2018-02-14 at 08:35 -0800, Paul Clarke wrote:
> @ThinkOpenly requested changes on this pull request.
> In src/testsuite/arith128_test_i128.c:
> > @@ -885,6 +885,20 @@ test_4b (void)
>  #endif
>    rc += check_vint256 ("vec_mul10ecuq:", l, k, (vui128_t) ec,
> (vui128_t) e);
>  
> +  return (rc);
> +}
> +
> +
> +/* Needed to split these tests into a separate function to avoid a
> ICE
> +   in GCC 6.3 / AT10.  */
> Do we know why splitting avoids the problem?

It is a gcc bug in the generation of debug info in GCC 5/6.

Bill does not think this is worth fixing for AT9/10 as there is work
around (compile without -g or make big functions smaller) and GCC 7
(AT11) does not have this bug.

> In src/testsuite/arith128_test_i128.c:
> > @@ -985,7 +999,6 @@ test_4b (void)
>  #endif
>    rc += check_vint256 ("vec_mul10ecuq:", l, k, (vui128_t) ec,
> (vui128_t) e);
>  
> -#if 1
> This is more than splitting, so why is this change included?

Just removing debug code. During debug I often disable sections of code
to isolate a problem, then reenable for final test. I tend to leave
these switches in in case I need them later :-/

> In src/testsuite/arith128_test_i128.h:
> > @@ -54,3 +57,4 @@ extern int
>  test_43 (void);
>  
>  #endif /* TEST_ARITH128_TEST_I128_H_ */
> +
> it's preferred to avoid unnecessary whitespace changes
> In src/testsuite/pveclib_test.c:
> > @@ -76,3 +77,4 @@ main (void)
>      printf ("%d failures reported\n", rc);
>    return (rc);
>  }
> +
> unnecessary whitespace change here, too
> —

I'll clean that up.

**Comment by ThinkOpenly** [comment]
On 02/14/2018 12:53 PM, Steven Munroe wrote:
> On Wed, 2018-02-14 at 08:35 -0800, Paul Clarke wrote:
>> @ThinkOpenly requested changes on this pull request.
>> In src/testsuite/arith128_test_i128.c:
>>> @@ -885,6 +885,20 @@ test_4b (void)
>> +/* Needed to split these tests into a separate function to avoid a
>> ICE
>> +   in GCC 6.3 / AT10.  */

>> Do we know why splitting avoids the problem?
> 
> It is a gcc bug in the generation of debug info in GCC 5/6.
> 
> Bill does not think this is worth fixing for AT9/10 as there is work
> around (compile without -g or make big functions smaller) and GCC 7
> (AT11) does not have this bug.

OK. Let's add some verbiage about this in the commit message as well.

>> In src/testsuite/arith128_test_i128.c:
>>> @@ -985,7 +999,6 @@ test_4b (void)

>> -#if 1
>> This is more than splitting, so why is this change included?
> 
> Just removing debug code. During debug I often disable sections of code
> to isolate a problem, then reenable for final test. I tend to leave
> these switches in in case I need them later :-/

Let's put this change in a separate commit that explains what's going on better, since it's not related to the function splitting going on in this commit.

Could you create a new branch in your clone, then generate your changes there:
first commit: function split
second commit: delete debug code

Then, submit a pull request for that branch?  OK?

**Comment by munroesj52** [comment]
So how do drop this pull request? "Close and comment" button?

**Comment by munroesj52** [comment]
from the documentation I can "Close and comment" without merging.

**Comment by munroesj52** [comment]
On Thu, 2018-02-15 at 15:00 +0000, Paul Clarke wrote:
> On 02/14/2018 12:53 PM, Steven Munroe wrote:
> > On Wed, 2018-02-14 at 08:35 -0800, Paul Clarke wrote:
> >> @ThinkOpenly requested changes on this pull request.
> >> In src/testsuite/arith128_test_i128.c:
> >>> @@ -885,6 +885,20 @@ test_4b (void)
> >> +/* Needed to split these tests into a separate function to avoid
> a
> >> ICE
> >> +   in GCC 6.3 / AT10.  */
> 
> >> Do we know why splitting avoids the problem?
> > 
> > It is a gcc bug in the generation of debug info in GCC 5/6.
> > 
> > Bill does not think this is worth fixing for AT9/10 as there is
> work
> > around (compile without -g or make big functions smaller) and GCC 7
> > (AT11) does not have this bug.
> 
> OK. Let's add some verbiage about this in the commit message as well.
> 
> >> In src/testsuite/arith128_test_i128.c:
> >>> @@ -985,7 +999,6 @@ test_4b (void)
> 
> >> -#if 1
> >> This is more than splitting, so why is this change included?
> > 
> > Just removing debug code. During debug I often disable sections of
> code
> > to isolate a problem, then reenable for final test. I tend to leave
> > these switches in in case I need them later :-/
> 
> Let's put this change in a separate commit that explains what's going
> on better, since it's not related to the function splitting going on
> in this commit.
> 
> Could you create a new branch in your clone, then generate your
> changes there:
> first commit: function split
> second commit: delete debug code.

This is really frustrating me (I am not a git ninja)

I do not know how to reset my fork to match the pveclib master and my
fork already has multiple commit messages embedded. So any branch I
create is likely to include the stuff you don't want.

The only thing I can think of is to destroy my fork and then recreate
the fork (If I can remember the twisty little passage that got me to
that whimsical point) and then branch.

**Comment by munroesj52** [comment]
On Wed, 2018-02-14 at 16:31 +0000, Paul Clarke wrote:
> I'm still learning here, but what I see in pull #4 are two commits,
> one titled "../COMMIT_EDITMSG-20180126", which has already been
> committed, and a second, "￼For now we should just split
> arith128_test_i128.c (test_4b) into two".
> I recommend creating new branches for your changes, creating pull
> requests for the branches, deleting the branches, then catching your
> master up (git fetch ...; git merge) after that pull request has been
> committed.
> —

Ok I destroyed then recreated my fork then created a branch (FixICE-
2018). 

To this branch I applied my patch and simplied it to just the required
changes. Then commited and pushed this branch and create a pull
request.

Hopefully this is clean and to you liking.

**Comment by ThinkOpenly** [comment]
Re: syncing a fork
https://help.github.com/articles/syncing-a-fork/

Then, from that clone's master, create a new branch and work there.  I delete branches when things are committed, and resync the master.  Rinse, repeat.  I'm just starting to get comfortable now, so you're not alone by any means.

**Comment by munroesj52** [comment]
On Fri, 2018-02-16 at 17:49 +0000, Paul Clarke wrote:
> Re: syncing a fork
> https://help.github.com/articles/syncing-a-fork/
> Then, from that clone's master, create a new branch and work there. I
> delete branches when things are committed, and resync the master.
> Rinse, repeat. I'm just starting to get comfortable now, so you're
> not alone by any means.
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub, or mute the thread.

This is way more complicated then you let on ...

$ git clone https://github.com/open-power-sdk/pveclib
Cloning into 'pveclib'...
remote: Counting objects: 156, done.
remote: Compressing objects: 100% (8/8), done.
remote: Total 156 (delta 0), reused 1 (delta 0), pack-reused 148
Receiving objects: 100% (156/156), 548.21 KiB | 0 bytes/s, done.
Resolving deltas: 100% (48/48), done.
Checking connectivity... done.
$ cd pveclib/
$ git fetch upstream
fatal: 'upstream' does not appear to be a git repository
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.

is "upstream" a thing or do I need to substitute my specific names.

reading further:

$ git remote -v
origin  https://github.com/munroesj52/pveclib (fetch)
origin  https://github.com/munroesj52/pveclib (push)
$ git remote add upstream https://github.com/open-power-sdk/pveclib
$ git remote -v
origin  https://github.com/munroesj52/pveclib (fetch)
origin  https://github.com/munroesj52/pveclib (push)
upstream        https://github.com/open-power-sdk/pveclib (fetch)
upstream        https://github.com/open-power-sdk/pveclib (push)
 
then:

$ git fetch upstream
remote: Counting objects: 1, done.
remote: Total 1 (delta 0), reused 0 (delta 0), pack-reused 0
Unpacking objects: 100% (1/1), done.
From https://github.com/open-power-sdk/pveclib
 * [new branch]      master     -> upstream/master
$ git checkout master
Already on 'master'
Your branch is up-to-date with 'origin/master'.
$ git merge upstream/master
Updating 62b2502..aa9426a
Fast-forward
 src/testsuite/arith128_test_i128.c | 14 ++++++++++++++
 src/testsuite/arith128_test_i128.h |  3 +++
 src/testsuite/pveclib_test.c       |  1 +
 3 files changed, 18 insertions(+)

so now the local clones of /open-power-sdk/pveclib and
munroesj52/pveclib match but has the remote of munroesj52/pveclib
changed?

of course not:

https://help.github.com/articles/pushing-to-a-remote/

$ git push origin master
Username for 'https://github.com': munroesj52
Password for 'https://munroesj52@github.com': 
Counting objects: 1, done.
Writing objects: 100% (1/1), 705 bytes | 0 bytes/s, done.
Total 1 (delta 0), reused 0 (delta 0)
To https://github.com/munroesj52/pveclib
   62b2502..aa9426a  master -> master

too many, steps very error prone (for me at least).

clearly I need to start writing my personal book of spells and
incantations ...

svn is simpler, makes more sense

**Comment by ThinkOpenly** [comment]
yes, forgot about the push back to sync your local clone with GitHub. having GitHub as the canonical repo for your clone adds that complication.  git is designed so that every repo is fairly standalone, and you tie them together with remotes, fetches, and pushes. it's powerful, but yeah, learning curve!
